eth-node/bridge/nimbus/node.go

```diff
@@ -18,46 +19,136 @@ import (
 	"runtime"
 	"strconv"
 	"strings"
+	"sync"
+	"syscall"
+	"time"
+	"unsafe"
 
 	"github.com/status-im/status-go/eth-node/crypto"
 	"github.com/status-im/status-go/eth-node/types"
 	enstypes "github.com/status-im/status-go/eth-node/types/ens"
 	"go.uber.org/zap"
 )
 
-func Init() {
-	runtime.LockOSThread()
+type nimbusNodeWrapper struct {
+	mu sync.Mutex
+
+	routineQueue     	*RoutineQueue
+	tid              	int
+	nodeStarted       bool
+	cancelPollingChan chan struct{}
+
+	w	types.Whisper
 }
 
-func StartNimbus(privateKey *ecdsa.PrivateKey, listenAddr string, staging bool) error {
-	C.NimMain()
+type Node interface {
+	types.Node
 
-	port, err := strconv.Atoi(strings.Split(listenAddr, ":")[1])
-	if err != nil {
-		return fmt.Errorf("failed to parse port number from %s", listenAddr)
+	StartNimbus(privateKey *ecdsa.PrivateKey, listenAddr string, staging bool) error
+	Stop()
+}
+
+func NewNodeBridge() Node {
+	c := make(chan Node, 1)
+	go func(c chan<- Node, delay time.Duration) {
+		runtime.LockOSThread()
+		defer runtime.UnlockOSThread()
+
+		n := &nimbusNodeWrapper{
+			routineQueue: 			NewRoutineQueue(),
+			tid: 								syscall.Gettid(),
```

Comment:
I see it weirdly indented, but I think it might be github interface

Reply:
There were some tabs there for some reason, fixed it.